Amazon.IonObjectMapper/IonSerializer.cs

```diff
@@ -327,6 +333,15 @@ public void Serialize<T>(IIonWriter writer, T item)
 
             if (item is object)
             {
+                var customSerializerAttribute = item.GetType().GetCustomAttribute<IonSerializerAttribute>();
+
+                if (customSerializerAttribute != null) {

```

Comment:
I think a private helper method handling the custom serializer creation logic (via factory or serializer type) would be an improvement.

Reply:
Added